lib/puma/server.rb

```diff
@@ -424,7 +425,7 @@ def process_client(client, buffer)
 
             ThreadPool.clean_thread_locals if clean_thread_locals
 
-            unless client.reset(@status == :run)
+            unless client.reset(@status == :run && @fast_track_ka_timeout)
```

Comment:
I added  `@fast_track_ka_timeout = fast_track_ka_timeout` inside the `run` method, but it's a mistake. I should have defined it in `initialize`. There isn't an accessor defined, but we could add it.